splice_machine/src/test/java/com/splicemachine/benchmark/execution/DeleteBenchmark.java

```diff
@@ -0,0 +1,171 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.junit.Assert.assertTrue;
+
+@Category(Benchmark.class)
+@RunWith(Parameterized.class)
+public class DeleteBenchmark extends ExecutionBenchmark {
+    private static final String SCHEMA = DeleteBenchmark.class.getSimpleName();
+
+    private final int numTableColumns;
+    private final int numIndexTables;
+    private final int tableSize;
+    private final boolean onOlap;
+
+    private final int batchSize;
+    private final String tableDefStr;
+    private final String indexDefStr;
+    private final String insertParamStr;
+
+    @ClassRule
+    public static SpliceSchemaWatcher spliceSchemaWatcher = new SpliceSchemaWatcher(SCHEMA);
+
+    public DeleteBenchmark(int numTableColumns, int numIndexColumns, int numIndexTables, int tableSize,
+        boolean onOlap) {
+        this.numTableColumns = Math.max(numTableColumns, 1);
+        this.numIndexTables = numIndexTables;
+        this.tableSize = tableSize;
+        this.onOlap = onOlap;
+        this.batchSize = Math.min(tableSize, 1000);
+
+        int realIndexColumns = Math.max(Math.min(numIndexColumns, numTableColumns), 1);
+
+        this.tableDefStr = getColumnDef(this.numTableColumns, TABLE_DEF);
+        this.indexDefStr = getColumnDef(realIndexColumns, INDEX_DEF);
+        this.insertParamStr = getColumnDef(this.numTableColumns, INSERT_PARAM);
+    }
+
+    static AtomicInteger curSize = new AtomicInteger(0);
+
+    @Before
+    public void setUp() throws Exception {
+        getInfo();
+
+        LOG.info("Create tables");
+        testConnection = makeConnection(spliceSchemaWatcher, false);
+        testStatement = testConnection.createStatement();
+        testStatement.execute("CREATE TABLE " + BASE_TABLE + tableDefStr);
+
+        createIndexes(numIndexTables, indexDefStr);
+
+        curSize.set(0);
+
+        testStatement.execute(String.format("call syscs_util.syscs_flush_table('%s', '%s')", SCHEMA, BASE_TABLE));
+        LOG.info("Collect statistics");
+        try (ResultSet rs = testStatement.executeQuery("ANALYZE SCHEMA " + SCHEMA)) {
+            assertTrue(rs.next());
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        testStatement.execute("DROP TABLE " + BASE_TABLE);
+        testStatement.close();
+        testConnection.close();
+    }
+
+    private void benchmark(boolean onOlap) {
+        String sqlText = String.format("DELETE FROM %s WHERE col_0 = ?",
+            BASE_TABLE);
+
+        try (Connection conn = makeConnection(spliceSchemaWatcher, onOlap)) {
+            try (PreparedStatement query = conn.prepareStatement(sqlText)) {
+                for (int i = 0; i < ExecutionBenchmark.NUM_EXECS; ++i) {
+                    curSize.set(0);
```

Comment:
I'd move _curSize_ to _ExecutionBenchmark_.
To initialize it, you may need to introduce
`protected void populateTables(SpliceSchemaWatcher spliceSchemaWatcher, int size, int batchSize, int numTableColumns, String insertParamStr, int concurrency)`
Make the current _populateTables_:
`private void populateTables(SpliceSchemaWatcher spliceSchemaWatcher, int size, int batchSize, int numTableColumns, String insertParamStr)`
Use _concurrency_ to either invoke it sequentially or via runBenchmark().